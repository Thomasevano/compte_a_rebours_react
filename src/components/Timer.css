.time-app {
  text-align: center;
  display: flex;
  height: 200px;
  width: 200px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  color: white;
}


.time {
  font-size: 50px;
}