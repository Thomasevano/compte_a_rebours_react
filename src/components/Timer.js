import React, { useState, useEffect } from 'react';
import './Timer.css';

const Timer = () => {
  const [seconds, setSeconds] = useState(40);
  const [isActive, setIsActive] = useState(false);

  function toggle() {
    setIsActive(!isActive);
  }

  function reset() {
    setSeconds(0);
    setIsActive(false);
  }

  function changeBackgroundColor () {
    if (40 >= seconds) {
      document.getElementById("countdown-app").style.background ="blue";
      if (20 >= seconds) {
        document.getElementById("countdown-app").style.background ="orange";
        if(12 >= seconds) {
          document.getElementById("countdown-app").style.background ="red";
        }
      }
    }
  }

  useEffect(() => {
    let interval = null;
    if (isActive) {
      interval = setInterval(() => {
        setSeconds(seconds => seconds - 1);
      }, 1000);
      if (seconds === 0){
        reset();
      }
      changeBackgroundColor();
    } else if (!isActive && seconds !== 0) {
      clearInterval(interval);
    }
    return () => clearInterval(interval);
  }, [isActive, seconds]);

  return (
    <div className="time-app" id="countdown-app">
      <div className="time">
        {seconds}s
      </div>
      <div className="row">
        <button onClick={toggle}>
          {isActive ? 'Pause' : 'Start'}
        </button>
      </div>
    </div>
  );
};

export default Timer;