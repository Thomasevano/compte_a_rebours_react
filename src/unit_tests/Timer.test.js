import React from 'react';
import Timer from '../components/Timer';
import renderer from 'react-test-renderer';

test('verify background color change according to the number of seconds displayed', () => {
  const component = renderer.create(
    <Timer>
    </Timer>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});