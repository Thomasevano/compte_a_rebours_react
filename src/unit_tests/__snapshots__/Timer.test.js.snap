// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify background color change according to the number of seconds displayed 1`] = `
<div
  className="time-app"
  id="countdown-app"
>
  <div
    className="time"
  >
    40
    s
  </div>
  <div
    className="row"
  >
    <button
      onClick={[Function]}
    >
      Start
    </button>
  </div>
</div>
`;
